docs/en/03_Upgrading/03_Migrating_off_CWP.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Migrating off CWP
+summary: Migrate your project off the Common Web Platform v2
+---
+
+# Migrate your project off the Common Web Platform v2
+
+Until November 2021, Silverstripe Ltd maintained a specialised version of the Silverstripe CMS project for the New Zealand Government and related New Zealand public-sector agencies. This version was called _Common Web Platform_ (CWP for short).
+
+The CWP CMS release has been discontinued. CWP projects who wish to continue to receive updates must migrate their code base off the latest CWP release back to the regular Silverstripe CMS release.
+
+## Steps to migcate away from CWP
+
+Before you begin this process you must identify which version of Silverstripe CMS Recipe you want to upgrade to. You can upgrade to [Silverstripe CMS Recipe 4.9.0](/changelogs/4.9.0/) or greater.
+
+[Review the changelog list](/changelogs/) to choose the Silverstripe CMS Recipe version you wish to upgrade to. Read the changelog carefully. It may contain additional release specific upgrade guidance beyond the steps in this guide.
+
+1. If your current composer file contains `cwp/cwp-installer`, run `composer update-recipe cwp/cwp-installer` to inline the recipe.
```

Comment:
I think these commands should be `composer require-recipe ...`, [as per the docs](https://github.com/silverstripe/recipe-plugin).

Reply:
I went through the doc and ran some local test. 

I think `composer require-recipe` is meant to install and inline a recipe that you have not yet installed in your project. If you call `composer require-recipe` for a recipe that you have already installed, nothing bad will happen. It will just inline the recipe, if it has not been inline already.

I think `composer udate-recipe` is meant to inline a recipe that you have already installed in your project. If you call `composer update-recipe` for a recipe that you NOT already installed, an error will be thrown. Arguably, this is a more desirable behaviour because it avoids the scenario of someone blindly following the instruction in the guide and accidentally installing a recipe.

TLDR Both `composer require-recipe` and `composer update-recipe` could work for this guide and will lead to the same outcome in most cases. `composer update-recipe` will however throw a potentially useful error if you try to inline a recipe you didn't mean to install.

Reply:
Ah, thanks for clarifying. I got that wrong :)